data/src/test/java/org/apache/iceberg/data/RandomGenericData.java

```diff
@@ -55,11 +55,9 @@ private RandomGenericData() {}
     return records;
   }
 
-  private static class RandomDataGenerator extends TypeUtil.CustomOrderSchemaVisitor<Object> {
-    private final Random random;
-
-    private RandomDataGenerator(long seed) {
-      this.random = new Random(seed);
+  private static class RandomRecordDataGenerator extends RandomDataGenerator<Record> {
```

Comment:
How about `RandomRecordGenerator` instead of `RandomRecordDataGenerator`? I think that's more descriptive since it returns random records.